api/v1beta1/azurecluster_webhook.go

```diff
@@ -33,27 +33,27 @@ import (
 func (c *AzureCluster) SetupWebhookWithManager(mgr ctrl.Manager) error {
 	return ctrl.NewWebhookManagedBy(mgr).
 		For(c).
+		WithDefaulter(&AzureCluster{}).
+		WithValidator(&AzureCluster{}).
```

Comment:
Still probing, but could setting this as empty structs be the issue ?

Reply:
Worth a try--I copied the empty struct pattern from CAPI without thinking about it much, but let's try using the calling structs instead.

Reply:
I made some progress on the tests with the commit https://github.com/kubernetes-sigs/cluster-api-provider-azure/pull/5563/commits/e3c1ffdba96da6103ea9b0c7424602329ee9d321 
My changes were inspired from https://github.com/kubernetes-sigs/cluster-api-provider-gcp/pull/1459/files